### DISCUSSION TITLE
Balta, no mapeamento das classes poderia ter usado o AutoMapper?

### DISCUSSION THREAD

**Discussion by joshuajka** [balta-io/forum#442] [Tira Dúvidas] [answered]
Se sim, existe algum vídeo seu mostrando a utilização do automapper para o mapeamento das entidades ?

**Comment by andrebaltieri** [comment, marked as answer]
@joshuajka bom dia, tudo bem?
Não tem... é bem difícil eu usar AutoMapper! Prefira Extension Methods ou Implicit/Explicit Operators:
👉 https://www.youtube.com/watch?v=2UVsT3TEVTw
👉 https://www.youtube.com/watch?v=xQhiv-bmt9o

**Reply by malighieri** [reply]
Uma alternativa para ter performance e ainda ter uma ferramenta que cuide da conversão/mapeamento pra você é o [Mapperly](https://github.com/riok/mapperly).
Ele é um gerador de código, ou seja, ele não vai usar reflection como o AutoMapper pra fazer a conversão em tempo de execução gerando maior consumo recursos e tempo. Ele vai, no momento de build, gerar um código que converte a Classe A para B, ou seja, fica bem próximo de você mesmo escrever a conversão usando implicit operators ou extension methods, com a vantagem que você não vai ter que ficar gerenciando as mudanças no código que afetam esta conversão. Neste vídeo são comparadas várias ferramentas de conversão, incluindo o AutoMapper, o Mapperly e até conversão manual.
[The Best .NET Mapper to Use in 2023 by Nick Chapsas](https://www.youtube.com/watch?v=U8gSdQN2jWI)
